Framework/Beamline/inc/MantidBeamline/ComponentInfo.h

```diff
@@ -158,11 +157,12 @@ class MANTID_BEAMLINE_DLL ComponentInfo {
                       const Eigen::Vector3d &scaleFactor);
   ComponentType componentType(const size_t componentIndex) const;
 
-  size_t scanCount(const size_t index) const;
+  size_t scanCount() const;
   size_t scanSize() const;
   bool isScanning() const;
-  std::pair<int64_t, int64_t>
-  scanInterval(const std::pair<size_t, size_t> &index) const;
+  const std::vector<std::pair<int64_t, int64_t>> &scanIntervals() const {
+    return m_scanIntervals;
+  }
```

Comment:
Avoid inlining functions unless there is a good performance reason for it. I think this should probably go in the `.cpp`.